### DISCUSSION TITLE
Any generic direct download link to latest version of GitExtensions ?

### DISCUSSION THREAD

**Discussion by Nataq666** [gitextensions/gitextensions#10519] [Q&A]
We built a way to automatically create a package (get all the files needed) for a SCCM deployment whenever we know a specific software has a new version. To do that, we either use a direct link to new setup file if that link is "generic" or we search a web page for links that has ".msi" (for exemple) so we can find and use that link to retrieve the setup file. This work fine, although we haven't been able to find a generic direct link to the latest version of GitExtensions .msi. Is there one ? Or any suggestion how to download the setup file without changing our code everytime to the new URL ? Thank you veru much !
Jeff

**Comment by pmiossec** [comment]
> or we search a web page for links that has ".msi" (for exemple) so we can find and use that link to retrieve the setup file.

Using this method, maybe you could use the link provided by github to the latest release and search for the msi link:
https://github.com/gitextensions/gitextensions/releases/latest

**Reply by Nataq666** [reply]
We tried. On that page, there is this link : https://github.com/gitextensions/gitextensions/releases/download/v4/GitExtensions-4.0.0-cba315df5.msi but it can't be found if we scan the page searching for ".msi" for exemple. It might be built dynamically (in parts) as far as I can tell.

**Reply by pmiossec** [reply]
Unfortunately if this page does not work, we have nothing more...

**Reply by mstv** [reply]
The page works (for me). The download link is correct. Perhaps the scraper script needs to execute some JS.

**Reply by cristianst85** [reply]
Sort of a hint, don't use "web scraping", but use the GitHub API instead to enumerate through the assets and find the appropriate link, etc.

Use this https://api.github.com/repos/GitExtensions/GitExtensions/releases/latest as a starting point, then enumerate the _assets_, match the _name_ based on whether it starts with _GitExtensions_ and ends with _.msi_ and then use the _browser_download_url_ as the final URL for download.

**Reply by pmiossec** [reply]
Really good point @cristianst85 
Your solution is there @Nataq666 ...

**Reply by Nataq666** [reply]
OK! The download link does show up in that URL so I should be able to do something with it! Trying to do this in powershell :

(Invoke-WebRequest "https://api.github.com/repos/GitExtensions/GitExtensions/releases/latest").Content | select-string -Pattern ".msi"

But then it returns the whole page content wich is interpreted as 1 big string as I understand it. Thinking about dumping the result in a text file and then try to search that file. Working on it !

Thanks again!

**Reply by vbjay** [reply]
Why they said don't scrape. The API call is the better way.

**Reply by vbjay** [reply]
https://blog.markvincze.com/download-artifacts-from-a-latest-github-release-in-sh-and-powershell/

**Comment by RussKie** [comment]
We have [this file](https://github.com/gitextensions/gitextensions/blob/configdata/GitExtensions.releases), which you could parse and get the latest link from.
